### ISSUE TITLE
The unlock/login interface keeps repeatedly popping up due to SSH signing requests

### ISSUE THREAD

**Issue by damnever** [strongbox-password-safe/Strongbox#717]
![strongbox-multiple-ssh](https://github.com/strongbox-password-safe/Strongbox/assets/6223594/656fe472-40a7-46c9-9b84-6178d3afcef5)

Some contexts from [#522](https://github.com/strongbox-password-safe/Strongbox/issues/522#issuecomment-1595604794) 
              
> > I use Vim with numerous plugins, all hosted on GitHub. However, whenever I upgrade these plugins simultaneously and the Strongbox is unfortunately locked during that time, the Strongbox unlock interface keeps popping up repeatedly, corresponding to the number of plugins. This becomes extremely annoying and renders the software virtually unusable.
>
>
> > Hi @damnever - Interesting, you could turn off approval messaging in the SSH Agent settings, this should be smoother. Would that work for you? I think longer term it would be good to say allow this process to use this key Always or something like that.        
>
> >@strongbox-mark I apologize for the confusion. I have already turned off the approval messaging because the messages were becoming too bothersome.
> >
> >Another case is: requesting the SSH key requires the database to be unlocked. If the database is locked, the unlock interface will keep popping up repeatedly too.
> 
>
> > Interesting in the second case, perhaps a setting, "Don't Ask for Approval If Database Locked - Just Fail"?
> 
> >
> > > requesting the SSH key requires the database to be unlocked. If the database is locked, the unlock/login interface will keep popping up repeatedly too.
> >
> > I am not familiar with desktop/macOS development. Perhaps there is a locking mechanism, such as file locking, that could be used?
> 
> 
> > Unfortunately I don't think so. The model we use is different from KeePassXC which passes your keys off to a separate SSH Agent that keeps them available when KeePassXC is not around or unlocked.
> >
> > In this case, the only thing I can think of if you need to have your keys always available is to keep your database unlocked. You can do this in the background without UI though.

**Comment by strongbox-mark** [comment]
I wonder, do you have multiple SSH keys (lets call the number of acceptable keys "n") stored in Strongbox that are accepted by the server you are trying to connect to?

If so, does Strongbox then prompt you n times? I think this is the only way this multiple request can happen...

Secondly, the issue is not with the first request which you try to cancel, but with subsequent requests, correct?

**Comment by damnever** [comment]
> do you have multiple SSH keys

Yes, I do have multiple SSH keys, however, the only one in use here should be the one for GitHub.

> does Strongbox then prompt you n times?

No, Strongbox prompts me far more than n times. I believe the number corresponds to how many `git pull` commands my vim plugin manager has executed.


> the issue is not with the first request which you try to cancel, but with subsequent requests, correct?

Yes, I have submitted the password and cancelled a lot of requests before recording the screen

**Comment by strongbox-mark** [comment]
Hi @damnever - OK, yeah I think that makes sense... 

In this case, what would you like to see happen when these other programs try to make those `git pull` commands? 

What would be the ideal behaviour if your Strongbox database is locked? Just fail the requests, no prompt?

**Comment by damnever** [comment]
@strongbox-mark Yes, indeed, rejecting the requests could be the solution, but I propose an alternative. I believe in this case, latency and concurrency aren't truly significant, so the socket server could queue/block the signing requests if the Strongbox database is locked. Once the unlocking action is completed, it should be up to Strongbox to determine whether all subsequent requests should pass or fail, or those requests timed out before the unlocking action was completed.

Another issue is that the Strongbox unlock/login interface continues to pop up, even if the signing requests are cancelled and the connections are closed.

**Comment by strongbox-mark** [comment]
Understood... Will look into an improvement here.

**Comment by strongbox-mark** [comment]
Hi @damnever - We've added a couple of minor improvements here in 1.58.20 - We try to throttle repeated requests, and you can also disable Unlock requests entirely (see the SSH Agent settings screen). Hopefully this will make things a lot easier for you.

**Comment by strongbox-mark** [comment]
That's fantastic news, but I don't quite understand what you mean when you say Strongbox prompts you again. 

Is the database locked or unlocked when you are prompted to unlock again? 

Would you be able to capture a screen recording and send to me at support@strongboxsafe.com?

**Comment by damnever** [comment]
I am unable to reproduce it, but I don't think that matters now. 

---

Nevertheless, I will leave a text version from memory: 
1. Git pull
2. Strongbox unlock is successful
3. Less than one minute 
4. I open the main window of Strongbox and click on the database
5. Strongbox prompts me with the unlock interface again.

**Comment by strongbox-mark** [comment]
OK, let's close this out for now, and feel free to open a new one if you can reproduce.
